src/wallet/wallet2.cpp

```diff
@@ -229,76 +229,74 @@ namespace {
-                os << "Votes are not stored in ascending order";
-
-            if (tx)
-                os << "TX Version: {}, Type: {}"_format(tx->version, tx->type);
-
-            std::string buf = os.str();
-            if (buf.size() >= 2 && buf[buf.size() - 2] == ',')
-                buf.resize(buf.size() - 2);
-
-            return buf;
-        }
+
+        std::ostringstream os;
+        assert(res.contains("tvc"));
+        const auto tvc = res["tvc"];
+
+        if (auto got = tvc.find("m_verbose_error"); got != tvc.end())
+            os << res["tvc"]["m_verbose_error"].get<std::string_view>() << "\n";
+        if (auto got = tvc.find("m_verifivation_failed"); got != tvc.end())
+            os << "Verification failed, connection should be dropped, ";  // bad tx, should drop
+                                                                          // connection
+        if (auto got = tvc.find("m_verifivation_impossible"); got != tvc.end())
+            os << "Verification impossible, related to alt chain, ";  // the transaction is
+                                                                      // related with an
+                                                                      // alternative blockchain
+        if (auto got = tvc.find("m_should_be_relayed"); got == tvc.end())
+            os << "TX should NOT be relayed, ";
+        if (auto got = tvc.find("m_added_to_pool"); got != tvc.end())
+            os << "TX added to pool, ";
+        if (auto got = tvc.find("m_low_mixin"); got != tvc.end())
+            os << "Insufficient mixin, ";
+        if (auto got = tvc.find("m_double_spend"); got != tvc.end())
+            os << "Double spend TX, ";
+        if (auto got = tvc.find("m_invalid_input"); got != tvc.end())
+            os << "Invalid inputs, ";
+        if (auto got = tvc.find("m_invalid_output"); got != tvc.end())
+            os << "Invalid outputs, ";
+        if (auto got = tvc.find("m_too_few_outputs"); got != tvc.end())
+            os << "Need at least 2 outputs, ";
+        if (auto got = tvc.find("m_too_big"); got != tvc.end())
+            os << "TX too big, ";
+        if (auto got = tvc.find("m_overspend"); got != tvc.end())
+            os << "Overspend, ";
+        if (auto got = tvc.find("m_fee_too_low"); got != tvc.end())
+            os << "Fee too low, ";
+        if (auto got = tvc.find("m_invalid_version"); got != tvc.end())
+            os << "TX has invalid version, ";
+        if (auto got = tvc.find("m_invalid_type"); got != tvc.end())
+            os << "TX has invalid type, ";
+        if (auto got = tvc.find("m_key_image_locked_by_snode"); got != tvc.end())
+            os << "Key image is locked by service node, ";
+        if (auto got = tvc.find("m_key_image_blacklisted"); got != tvc.end())
+            os << "Key image is blacklisted on the service node network, ";
+
+        const auto m_vote_ctx = tvc["m_vote_ctx"];
```

Comment:
Is this value always present?

Reply:
Not too sure, the wallet is very broken since it switched over to using JSON. I have a suspicion that potentially all the code paths are generating "reason" irrespective of the blink flag being set or not. Sending an ONS transaction triggers the non-blink path but still emitted a "reason".